consensus/ibft/src/test/java/tech/pegasys/pantheon/consensus/ibft/jsonrpc/methods/IbftGetSignerMetricsTest.java

```diff
@@ -0,0 +1,269 @@
+
+import tech.pegasys.pantheon.consensus.ibft.IbftBlockInterface;
+import tech.pegasys.pantheon.consensus.ibft.IbftExtraData;
+import tech.pegasys.pantheon.consensus.ibft.IbftExtraDataFixture;
+import tech.pegasys.pantheon.consensus.ibft.Vote;
+import tech.pegasys.pantheon.crypto.SECP256K1;
+import tech.pegasys.pantheon.ethereum.core.Address;
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.core.BlockHeaderTestFixture;
+import tech.pegasys.pantheon.ethereum.core.Hash;
+import tech.pegasys.pantheon.ethereum.core.Util;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.queries.BlockchainQueries;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.SignerMetricResult;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+import tech.pegasys.pantheon.util.uint.UInt256;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.LongStream;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class IbftGetSignerMetricsTest {
+
+  private final JsonRpcParameter jsonRpcParameter = new JsonRpcParameter();
+  private final String IBFT_METHOD = "ibft_getSignerMetrics";
+  private final String JSON_RPC_VERSION = "2.0";
+  private IbftGetSignerMetrics method;
+
+  private BlockchainQueries blockchainQueries;
+  private IbftBlockInterface ibftBlockInterface;
+
+  @Rule public ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setup() {
+    blockchainQueries = mock(BlockchainQueries.class);
+    ibftBlockInterface = mock(IbftBlockInterface.class);
+    method = new IbftGetSignerMetrics(ibftBlockInterface, blockchainQueries, jsonRpcParameter);
+  }
+
+  @Test
+  public void returnsCorrectMethodName() {
+    assertThat(method.getName()).isEqualTo(IBFT_METHOD);
+  }
+
+  @Test
+  public void exceptionWhenInvalidStartBlockSupplied() {
+    final JsonRpcRequest request = requestWithParams("INVALID");
+
+    expectedException.expect(InvalidJsonRpcParameters.class);
+    expectedException.expectMessage("Invalid json rpc parameter at index 0");
+
+    method.response(request);
+  }
+
+  @Test
+  public void exceptionWhenInvalidEndBlockSupplied() {
+    final JsonRpcRequest request = requestWithParams("1", "INVALID");
+
+    expectedException.expect(InvalidJsonRpcParameters.class);
+    expectedException.expectMessage("Invalid json rpc parameter at index 1");
+
+    method.response(request);
+  }
+
+  @Test
+  public void getSignerMetricsWhenNoParams() {
+
+    Long startBlock = 1L;
+    Long endBlock = 3L;
```

Comment:
nit: These should just be primitives and we add the `final` keyword whenever possible in Pantheon (well try to - we're not as consistent as we should be).  So:
```suggestion
    final long endBlock = 3L;
```

Reply:
done